### PR TITLE
CPM-971: Add CategoriesSelector

### PR DESCRIPTION
**Pull request by pierallard**

### CODE REVIEW

File: components/identifier-generator/front/src/feature/components/CategoryTreeSwitcher.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+  return (
+    <Dropdown>
+      <SwitcherButton label={'Tree'} onClick={open}>
+        {trees.find(tree => tree.code === value)?.label}
+      </SwitcherButton>
+      {isOpen && (
+        <Dropdown.Overlay onClose={close}>
+          <Dropdown.Header>
+            <Dropdown.Title>Categories</Dropdown.Title>
```
Comment (mmetayer) [inline components/identifier-generator/front/src/feature/components/CategoryTreeSwitcher.tsx:28]: translate

File: components/identifier-generator/front/src/feature/components/CategoriesSelector.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        value={categoryCodes}
+        onChange={onChange}
+        onFocus={open}
+        labels={filledCategoryLabels}
+        invalidValue={invalidValue}
+      />
+      {isOpen && (
+        <Dropdown.Overlay onClose={close} horizontalPosition={'left'}>
+          <Dropdown.Header>
+            <Dropdown.Title>Categories</Dropdown.Title>
```
Comment (mmetayer) [inline components/identifier-generator/front/src/feature/components/CategoriesSelector.tsx:60]: translate

File: front-packages/akeneo-design-system/src/components/Input/TagInput/TagInput.tsx [hunk 3]
```diff
@@ -116,6 +119,16 @@ type TagInputProps = Override<
...
     onSubmit?: () => void;
+
+    /**
+     * Displays tag labels instead of tags
+     */
+    labels?: {[key: string]: string};
+
+    /**
+     * The selected tags which are invalid
+     * */
```
Comment (mmetayer) [inline front-packages/akeneo-design-system/src/components/Input/TagInput/TagInput.tsx:130]:
```suggestion
     */
```

**Review by mmetayer — APPROVED** [review]

File: components/identifier-generator/front/src/feature/hooks/useCategoryTrees.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+            selected: 'true' | 'false';
+            tree?: CategoryTreeModel;
+          }) => {
+            return {...tree, selected: tree.selected === 'true'};
+          }
+        );
+        setTrees(trees);
+        const currentTree = trees.find(tree => tree.selected);
+        if (currentTree) {
+          callback(currentTree);
```
Comment (cemorin) [inline components/identifier-generator/front/src/feature/hooks/useCategoryTrees.ts:35]: What it's for? In the case with the `onChange` for exemple

File: components/identifier-generator/front/src/feature/hooks/useCategoryTrees.ts [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        }
+      });
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [callback]);
```
Comment (cemorin) [inline components/identifier-generator/front/src/feature/hooks/useCategoryTrees.ts:40]: Why the disable? Which param relaunches it?

**Review by cemorin — APPROVED** [review]

File: components/identifier-generator/front/src/feature/components/CategoryTreeSwitcher.tsx [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+  const handleChange = (tree: CategoryTreeRoot) => {
+    close();
+    onChange(tree);
+  };
+
+  return (
+    <Dropdown>
+      <SwitcherButton label={'Tree'} onClick={open}>
+        {trees.find(tree => tree.code === value)?.label}
```
Comment (cemorin) [inline components/identifier-generator/front/src/feature/components/CategoryTreeSwitcher.tsx:23]:
```suggestion
        {trees.find(({code}) => code === value)?.label}
```

File: components/identifier-generator/front/src/feature/components/CategoryTreeSwitcher.tsx [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+  const trees = useCategoryTrees(onChange);
+
+  const handleChange = (tree: CategoryTreeRoot) => {
+    close();
+    onChange(tree);
+  };
+
+  return (
+    <Dropdown>
+      <SwitcherButton label={'Tree'} onClick={open}>
```
Comment (cemorin) [inline components/identifier-generator/front/src/feature/components/CategoryTreeSwitcher.tsx:22]: This label is never showed right?

File: components/identifier-generator/front/src/feature/hooks/useCategoryTrees.ts [hunk 8]
```diff
@@ -0,0 +1,45 @@
+import {useState, useEffect} from 'react';
+import {CategoryTreeCode, CategoryTreeModel, CategoryTreeRoot, useRouter} from '@akeneo-pim-community/shared';
+
+const useCategoryTrees: (callback: (categoryTreeRoot: CategoryTreeRoot) => void) => CategoryTreeRoot[] = callback => {
+  const router = useRouter();
+  const [trees, setTrees] = useState<CategoryTreeRoot[]>([]);
```
Comment (cemorin) [inline components/identifier-generator/front/src/feature/hooks/useCategoryTrees.ts:6]: Don't we need `loading` information in order to show some kind of skeleton somewhere?

File: components/identifier-generator/front/src/feature/components/CategoryTreeSwitcher.tsx [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+        <Dropdown.Overlay onClose={close}>
+          <Dropdown.Header>
+            <Dropdown.Title>Categories</Dropdown.Title>
+          </Dropdown.Header>
+          <Dropdown.ItemCollection>
+            {trees.map(tree => (
+              <Dropdown.Item isActive={tree.code === value} key={tree.code} onClick={() => handleChange(tree)}>
+                {tree.label}
+              </Dropdown.Item>
+            ))}
```
[the remark this answers is not in the document]
Comment (pierallard) [reply components/identifier-generator/front/src/feature/components/CategoryTreeSwitcher.tsx:35]: Nope, I need "tree" :)

File: src/Akeneo/Pim/Structure/Component/Query/PublicApi/Category/GetCategoryTranslations.php [hunk 10]
```diff
@@ -6,5 +6,8 @@
 
 interface GetCategoryTranslations
 {
+    /**
+     * @return array<string, string>
+     */
```
[the remark this answers is not in the document]
Comment (mmetayer) [reply src/Akeneo/Pim/Structure/Component/Query/PublicApi/Category/GetCategoryTranslations.php:11]: BTW, not related to this PR

File: components/identifier-generator/front/src/feature/hooks/useCategoryTrees.ts (hunk 8 above)
Comment (pierallard) [reply components/identifier-generator/front/src/feature/hooks/useCategoryTrees.ts:6]: There is already a loader in the tree. If no tree is defined, there is a loading icon.

Comment (pierallard) [reply components/identifier-generator/front/src/feature/hooks/useCategoryTrees.ts:6]: There is already a loader in the tree. If no tree is defined, there is a loading icon.

**Pull request merged**